// containers/App.js
import React, { Component } from 'react';
import {connect} from 'react-redux'

import {getFlightsAll} from 'selectors'
import {getFlightsArriv} from 'selectors'
import {getFlightsDepart} from 'selectors'
import {getFlightsDel} from 'selectors'

class App extends Component {
  constructor(){
      super()
      this.state = {
          fl: []
      }

  }

  row (flight, index) {
      return (
              <div className="row" key={index}>
                  <div className="col-sm">{flight.time}</div>
                  <div className="col-sm">{flight.name}</div>
                  <div className="col-sm">{flight.number}</div>
                  <div className="col-sm">{flight.status}</div>
              </div>
      )
  }

  componentDidMount(){
    this.select(this.props.flightsAll)
  }

  select(pr) {
    this.setState({fl: pr})
  }

  render() {
    return (
      <div>
          <div className="container">
              <hr/>
              <h2>Wellcome to Moscow !</h2>
              <hr/>
              <h4>Current Time is {this.props.currentTime}</h4>

              <button onClick={this.select.bind(this, this.props.flightsAll)}>All flights</button>
              <button onClick={this.select.bind(this, this.props.flightsArriv)}>Arrivals</button>
              <button onClick={this.select.bind(this, this.props.flightsDepart)}>Departures</button>
              <button onClick={this.select.bind(this, this.props.flightsDel)}>Delayed</button>

              <hr/>
              <h6>put the number of flight for searching</h6>
              <input type="text"/>
              <br/> available numbers for searching are 1000, 2000, 3000, 4000

              <hr/>
              <div className="container">
                 <div className="row">
                    <div className="col-sm">Time</div>
                    <div className="col-sm">Direction</div>
                    <div className="col-sm">Number</div>
                    <div className="col-sm">Status</div>
                 </div>
              </div>

              <hr/>
              <div className="container">
                  {this.state.fl.map((flight, index) => this.row(flight, index))}
              </div>

              <div className="red">
                No one was found.
                <br/>  Please put number of flight more carefully.
              </div>

          </div>
      </div>
    );
  }
}

const mSTP = (st) => ({
    flightsAll: getFlightsAll(st),
    flightsArriv: getFlightsArriv(st),
    flightsDepart: getFlightsDepart(st),
    flightsDel: getFlightsDel(st),
    currentTime: st.currentTime

})

export default connect(mSTP)(App)
